src/environmentd/src/http/sql.rs

```diff
@@ -101,6 +106,147 @@ impl Error {
     }
 }
 
+static PER_REPLICA_LABELS: &[&str] = &["replica_full_name", "instance_id", "replica_id"];
+
+async fn execute_promsql_query(
+    client: &mut AuthedClient,
+    query: &PrometheusSqlQuery<'_>,
+    metrics_registry: &MetricsRegistry,
+    metrics_by_name: &mut BTreeMap<String, GenericGaugeVec<AtomicF64>>,
+    cluster: Option<(&Cluster, &ClusterReplica)>,
+) {
+    assert!(query.per_replica == cluster.is_some());
+
+    let mut res = SqlResponse {
+        results: Vec::new(),
+    };
+
+    execute_request(client, query.to_sql_request(cluster), &mut res)
+        .await
+        .expect("valid SQL query");
+
+    let result = match res.results.as_slice() {
+        [SqlResult::Ok { .. }, SqlResult::Ok { .. }, SqlResult::Ok { .. }, result] => result,
```

Comment:
It's not really clear to my why we're always expecting and ignoring three `SqlResult::Ok`.